components/graph/graph.js

```diff
@@ -219,9 +226,15 @@ class GraphViewModel {
     return edge;
   }
 
-  markNodesIdeologicalBranches(refs, nodes, nodesById) {
-    refs = refs.filter((r) => !!r.node());
-    refs = refs.sort((a, b) => {
+  markNodesIdeologicalBranches(refs) {
+    const refNodeMap = {};
+    refs.forEach((r) => {
+      if (!r.node()) return;
+      if (!r.node().timestamp) return;
+      if (refNodeMap[r.node().sha1]) return;
+      refNodeMap[r.node().sha1] = r;
```

Comment:
Does this actually speed up things? Are there that many duplicates that the overhead of deduplicating is worth it?